src/komodo_notary.cpp

```diff
@@ -299,38 +297,42 @@ struct notarized_checkpoint *komodo_npptr_for_height(int32_t height, int *idx)
         }
     }
     *idx = -1;
-    return(0);
+    return nullptr;
 }
 
-struct notarized_checkpoint *komodo_npptr(int32_t height)
+notarized_checkpoint *komodo_npptr(int32_t height)
 {
     int idx;
     return komodo_npptr_for_height(height, &idx);
 }
 
-struct notarized_checkpoint *komodo_npptr_at(int idx)
+notarized_checkpoint *komodo_npptr_at(int idx)
 {
-    char symbol[KOMODO_ASSETCHAIN_MAXLEN],dest[KOMODO_ASSETCHAIN_MAXLEN]; struct komodo_state *sp;
+    char symbol[KOMODO_ASSETCHAIN_MAXLEN];
+    char dest[KOMODO_ASSETCHAIN_MAXLEN]; 
+    komodo_state *sp;
     if ( (sp= komodo_stateptr(symbol,dest)) != 0 )
-        if (idx < sp->NUM_NPOINTS)
+        if (idx < sp->NPOINTS.size())
             return &sp->NPOINTS[idx];
-    return(0);
+    return nullptr;
 }
 
 int32_t komodo_prevMoMheight()
 {
     static uint256 zero;
-    char symbol[KOMODO_ASSETCHAIN_MAXLEN],dest[KOMODO_ASSETCHAIN_MAXLEN]; int32_t i; struct komodo_state *sp; struct notarized_checkpoint *np = 0;
+    char symbol[KOMODO_ASSETCHAIN_MAXLEN];
+    char dest[KOMODO_ASSETCHAIN_MAXLEN]; 
+    komodo_state *sp; 
+
     if ( (sp= komodo_stateptr(symbol,dest)) != 0 )
     {
-        for (i=sp->NUM_NPOINTS-1; i>=0; i--)
+        for (auto &pt : sp->NPOINTS)
         {
-            np = &sp->NPOINTS[i];
-            if ( np->MoM != zero )
-                return(np->notarized_height);
+            if ( pt.MoM != zero )
+                return pt.notarized_height;
         }
```

Comment:
It seems wrong order of `NPOINTS[i]` scan here. It was _from end to begin_ with exit when `MoM` is not zero, but you did it vice-versa order - _from begin to end_. I guess better to revert range-loop changes here to something like:
```
for (int32_t i=sp->NPOINTS.size()-1; i>=0; i--)
{
    if ( sp->NPOINTS[i].MoM != zero )
        return sp->NPOINTS[i].notarized_height;
}
```
Or use reverse iterators ... or range-loop with `boost::adaptors::reverse`.

Reply:
**p.s.** I vote for revert https://github.com/KomodoPlatform/komodo/pull/476/commits/de6166bddd3ab27e501b28dba5bbbe795c3251c7 from this PR.

Reply:
Agree. I attempted to knock it out too quickly. I'll move that to another branch. That'll give me more time to see what I can do about that last_NPOINTSi thing.